### PR TITLE
replace web3-utils methods usage by status-go

### LINKED ISSUE #13888: replace web3-utils methods usage by status-go
we want to have methods used from  web3-utils  moved(replaced) to staus-go

and remove web3-utils dependency

### PR DESCRIPTION
**Pull request by qfrank**
fixes #13888 

NOTE: this PR contains modifications from #13858

### CODE REVIEW

**Comment by churik** [comment]
@qfrank 
please, rebase it and request review while ready, thank you!

**Comment by qfrank** [comment]
ready to test @churik @flexsurfer :)

File: status-go-version.json
Comment (yqrashawn) [inline status-go-version.json]: We should depend on `status-im/status-go`
Seems this is blocked by https://github.com/status-im/status-go/pull/2840

**Review by yqrashawn — DISMISSED** [review]

File: status-go-version.json
Comment (qfrank) [reply status-go-version.json]: yeah, but it's a temp solution, will update after QA test pass

**Comment by flexsurfer** [comment]
let's test this PR first

**Review by flexsurfer — APPROVED** [review]

**Comment by pavloburykh** [comment]
> let's test this PR first

@flexsurfer hi! Just for clarifying, is this PR can be assumed as ready for testing? Should we ignore the last changing request (and the fact that PR has no approvals from reviewers)? Also, I see that there are unresolved conflicts in status go. 
**cc** @qfrank

**Comment by flexsurfer** [comment]
QA please ignore conflicts, they are not related

**Comment by pavloburykh** [comment]
> QA please ignore conflicts, they are not related

@flexsurfer got it, thanx! Could you please also provide a short QA notes of what should be tested in terms of these changes?

**Comment by flexsurfer** [comment]
@pavloburykh changes related to password, so create account, login, and also to wallet, send txs , interact with dapps
, it could be tested only in testnet for now,

**Comment by qfrank** [comment]
Hi @pavloburykh , i just rebased the PR to the latest develop :)

**Comment by pavloburykh** [comment]
@qfrank thanx for the PR. Please, take a look at the following issue. This issue is not reproducible in develop branch. It used to reproduce in the past, but couple of weeks ago it was fixed by https://github.com/status-im/status-mobile/pull/14074 Maybe current PR still misses some changes from develop or there is another problem. 

### ISSUE 1 Transaction's details in Status app (values of Trans fee and Gas price) differs from the ones on Etherscan

**Steps:** 

1. Generate any transaction in Status app (or use existing one from history of transactions)
2. Open transaction details in Status app (Go to Wallet tab - open Wallet screen - open history tab - select transaction from the list)
3. Open the details of the same transaction on Etherscan
4. Compare data from details page in Status app and Etherscan
5. Check if Transaction fee and Gas price values are the same

**Actual result:** there is a difference in values

**Expected result:** all data should be the same

photo_2022-10-12 17 37 45

photo_2022-10-12 17 37 47

**Comment by flexsurfer** [comment]
probably status-go should be rebased as well

**Comment by qfrank** [comment]
Hi @pavloburykh , thanks for reporting! Just solved issue 1 :)

**Comment by pavloburykh** [comment]
> Hi @pavloburykh , thanks for reporting! Just solved issue 1 :)

Thank you @qfrank ! Could you please take a look at integration test build failures? File with logs is attached. 
[log.txt](https://github.com/status-im/status-mobile/files/9772450/log.txt)

**Comment by qfrank** [comment]
Looks like there is something relate to @yqrashawn , could u help to solve it, thanks @yqrashawn 
```
[2022-10-13T02:11:58.305Z] FAIL in (delete-for-me) (status_im/chat/models/delete_message_for_me_test.cljs:24:13)
[2022-10-13T02:11:58.305Z] delete for me
[2022-10-13T02:11:58.305Z] expected: (= (delete-message-for-me/delete {:db db} message 1000) expected)
[2022-10-13T02:11:58.306Z]   actual: (not (= {:db {:messages {"chat-id" {"message-id" {:id "message-id", :deleted-for-me? true, :deleted-for-me-undoable-till 1665627119168}}}}, :utils/dispatch-later [{:dispatch [:chat.ui/delete-message-for-me-and-sync {:chat-id "chat-id", :message-id "message-id"}], :ms 1000}]} {:db {:messages {"chat-id" {"message-id" {:id "message-id", :deleted-for-me? true, :deleted-for-me-undoable-till 1665627119167}}}}, :utils/dispatch-later [{:dispatch [:chat.ui/delete-message-for-me-and-sync {:chat-id "chat-id", :message-id "message-id"}], :ms 1000}]}))
```

**Comment by yqrashawn** [comment]
@qfrank @pavloburykh The failed test is not related to changes in this PR. 
I've created https://github.com/status-im/status-mobile/issues/14155 for it.

**Comment by pavloburykh** [comment]
@qfrank thank you. PR looks good to me. Except a bug that was caught by E2E. The bug is not PR related but likely introduced by new go version. I am reproducing this bug in another PR https://github.com/status-im/status-mobile/pull/14144 that is also in testing. This bug is not in develop yet. Since then, merging is blocked until we figure out which of developers should fix this bug. 

Bug description:
### Edited message does not respond to long-press action - options menu does not appear. No mark in message bubble about this message was edited is displayed. 

**Steps:**

1. Send a message in any kind of chat
2. Edit message (see if mark "edited" is displayed in message bubble)
3. Try to open open options menu by long press on the edited message
4. Observe the result

**Actual result:** no mark "edited" is displayed in message bubble. Long-press on edited message does not result in options menu opening.


https://user-images.githubusercontent.com/97245802/195633819-4b592e53-9cf8-4d68-9c6a-36a293d284df.mp4

**Comment by flexsurfer** [comment]
definitely not related to this PR

**Comment by flexsurfer** [comment]
@pavloburykh i would suggest to merge this PR wdyt ?

**Comment by pavloburykh** [comment]
> @pavloburykh i would suggest to merge this PR wdyt ?

PR is ready for merge. Thnax @qfrank @flexsurfer

**Comment by flexsurfer** [comment]
hey @qfrank so you can now merge status-go branch, and then change version in this PR and merge it! thank you

File: status-go-version.json
Comment (flexsurfer) [inline status-go-version.json]: 0.111.5 should be used here

**Pull request merged**
